<a href="https://colab.research.google.com/github/AbrahamMH19/Modelos-de-Regresi-n-Lineal/blob/main/Actividad_5_Regresion_Lineal_A01737333.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Importar librerias
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from turtle import color

In [ ]:
# Subir base de Datos
df = pd.read_csv('cuentas_credicel.csv', encoding='latin')

df.head(5)

In [ ]:
df.info()

**Limpiar columnas**

In [13]:
# Eliminar el signo "S" de una columna "Plazo"
#df['plazo'] = df['plazo'].str.replace('S', '')
#df['plazo'] = df['plazo'].astype(int)

# Sustituirlos strings Atraso, Cancelado y Fraude por la constante "0"
df['riesgo']=df['riesgo'].str.replace('Atraso', '0')
df['riesgo']=df['riesgo'].str.replace('Cancelado', '0')
df['riesgo']=df['riesgo'].str.replace('Fraude', '0')
df["riesgo"]= df["riesgo"].fillna(0)
#Conversión de tipo de dato de columna de tipo Object a int
df['costo_total']= df['costo_total'].astype(float)

#Sustituyo valores nulos por el numero "0"
#Tercer método de sustitución de valores nulos
#Sustituir valores nulos por un valor numérico en  concreto
df["porc_enganche"]= df["porc_enganche"].fillna(0)

**Sacar Gráficas**

In [ ]:
sns.scatterplot(x='enganche', y='riesgo', color="red", data=df)

In [ ]:
sns.scatterplot(x='precio', y='plazo', color="red", data=df)

In [ ]:
sns.scatterplot(x='costo_total', y='monto_financiado', color="red", data=df)

In [ ]:
sns.scatterplot(x='pagos_realizados', y='porc_enganche', color="red", data=df)

In [19]:
# Definir modelo de regresion lineal
from sklearn.linear_model import LinearRegression
model = LinearRegression()

# Y declarar variables dependientes e independientes para la regresión lineal

**Modelo 1**

In [ ]:
# Modelo Matematico 1
Vars_Indep1= df[["enganche"]]
Var_Dep1= df["riesgo"]
model.fit(X=Vars_Indep1, y=Var_Dep1)
model.__dict__

In [21]:
# Coeficiente Determinacion 1
coef_Deter1 = model.score(X = Vars_Indep1, y = Var_Dep1)
coef_Deter1

0.042564202425108943

In [22]:
# Coeficiente Correlacion 1
coef_Correl1 = np.sqrt(coef_Deter1)
coef_Correl1

0.20631093627122374

**Modelo 2**

In [ ]:
# Modelo Matematico 2
Vars_Indep2= df[["precio"]]
Var_Dep2= df["plazo"]
model.fit(X=Vars_Indep2, y=Var_Dep2)
model.__dict__

In [24]:
# Coeficiente Determinacion 2
coef_Deter2 = model.score(X = Vars_Indep2, y = Var_Dep2)
coef_Deter2

0.041877006763330726

In [25]:
# Coeficiente Correlacion 1
coef_Correl2 = np.sqrt(coef_Deter2)
coef_Correl2

0.20463872254128915

**Modelo 3**

In [26]:
# Modelo Matematico 3
Vars_Indep3= df[["costo_total"]]
Var_Dep3= df["monto_financiado"]
model.fit(X=Vars_Indep3, y=Var_Dep3)
model.__dict__

{'fit_intercept': True,
 'copy_X': True,
 'n_jobs': None,
 'positive': False,
 'feature_names_in_': array(['costo_total'], dtype=object),
 'n_features_in_': 1,
 'coef_': array([0.42741682]),
 'rank_': 1,
 'singular_': array([359751.03527969]),
 'intercept_': 639.9922581004103}

In [27]:
# Coeficiente Determinacion 3
coef_Deter3 = model.score(X = Vars_Indep3, y = Var_Dep3)
coef_Deter3

0.8646380417764322

In [28]:
# Coeficiente Correlacion 3
coef_Correl3 = np.sqrt(coef_Deter3)
coef_Correl3

0.929859151579653

**Modelo 4**

In [29]:
# Modelo 4
Vars_Indep4= df[["pagos_realizados"]]
Var_Dep4= df["porc_enganche"]
model.fit(X=Vars_Indep4, y=Var_Dep4)
model.__dict__

{'fit_intercept': True,
 'copy_X': True,
 'n_jobs': None,
 'positive': False,
 'feature_names_in_': array(['pagos_realizados'], dtype=object),
 'n_features_in_': 1,
 'coef_': array([-0.15132576]),
 'rank_': 1,
 'singular_': array([1463.82068582]),
 'intercept_': 2.3432248342563584}

In [30]:
# Coeficiente Determinacion 4
coef_Deter4 = model.score(X = Vars_Indep4, y = Var_Dep4)
coef_Deter4

0.06523723314088559

In [31]:
# Coeficiente Correlacion 4
coef_Correl4 = np.sqrt(coef_Deter4)
coef_Correl4

0.25541580440702094

- - -
**Los modelos son los siguientes:**

---
             
*Modelo Matemático 1:* 0.00601946x - 3.422566722440304

*Modelo Determinación 1:* 0.042564202425108943

*Modelo Correlación 1:* 0.20631093627122374 texto en cursiva

---

Modelo Matemático 2: 0.00150129x + 19.90894959705299

*Modelo Determinación 2:* 0.041877006763330726

*Modelo Correlación 2:* 0.20463872254128915

---

Modelo Matemático 3: 0.42741682x + 639.9922581004103

*Modelo Determinación 3:* 0.8646380417764322

*Modelo Correlación 3:* 0.929859151579653

---

*Modelo Matemático 4:* -0.15132576x + 2.3432248342563584

*Modelo Determinación 4:* 0.06523723314088559

*Modelo Correlación 4:* 0.25541580440702094

---